tests/phpunit/tests/term/cache.php

```diff
@@ -219,4 +219,167 @@ public function test_term_objects_should_not_be_modified_by_update_term_cache()
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+
+		// This should now hit cache.
+		$term = get_term_by( 'slug', 'burrito', 'post_tag' );
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+
+		$this->assertEquals( get_term( $term_id, 'post_tag' ), $term );
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+	}
+
+	/**
+	 * @ticket 21760
+	 */
+	function test_get_term_by_slug_cache_update() {
+		global $wpdb;
+
+		$term_id = $this->factory->term->create( array( 'slug' => 'burrito', 'taxonomy' => 'post_tag' ) );
+
+		clean_term_cache( $term_id, 'post_tag' );
+		$num_queries = $wpdb->num_queries;
+
+		get_term_by( 'slug', 'burrito', 'post_tag' );
+		$num_queries++;
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+
+		// This should now hit cache.
+		get_term_by( 'slug', 'burrito', 'post_tag' );
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+
+		// Update the tag which invalidates the cache.
+		wp_update_term( $term_id, 'post_tag', array( 'name' => 'Taco' ) );
+		$num_queries = $wpdb->num_queries;
+
+		// This should not hit cache.
+		get_term_by( 'slug', 'burrito', 'post_tag' );
+		$num_queries++;
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+	}
+
+	/**
+	 * @ticket 21760
+	 */
+	function test_get_term_by_name_cache() {
+		global $wpdb;
+
+		$term_id = $this->factory->term->create( array( 'name' => 'burrito', 'slug' => 'noburrito', 'taxonomy' => 'post_tag' ) );
+
+		clean_term_cache( $term_id, 'post_tag' );
+		$num_queries = $wpdb->num_queries;
+
+		get_term_by( 'name', 'burrito', 'post_tag' );
+		$num_queries++;
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+
+		// This should now hit cache.
+		$term = get_term_by( 'name', 'burrito', 'post_tag' );
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+
+		$this->assertEquals( get_term( $term_id, 'post_tag' ), $term );
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+	}
+
+	/**
+	 * @ticket 21760
+	 */
+	function test_get_term_by_name_cache_update() {
+		global $wpdb;
+
+		$term_id = $this->factory->term->create( array( 'name' => 'burrito', 'slug' => 'noburrito', 'taxonomy' => 'post_tag' ) );
+
+		clean_term_cache( $term_id, 'post_tag' );
+		$num_queries = $wpdb->num_queries;
+
+		get_term_by( 'name', 'burrito', 'post_tag' );
+		$num_queries++;
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+
+		// This should now hit cache.
+		get_term_by( 'name', 'burrito', 'post_tag' );
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+
+		// Update the tag which invalidates the cache.
+		wp_update_term( $term_id, 'post_tag', array( 'slug' => 'taco' ) );
+		$num_queries = $wpdb->num_queries;
+
+		// This should not hit cache.
+		get_term_by( 'name', 'burrito', 'post_tag' );
+		$num_queries++;
+		$this->assertEquals( $num_queries, $wpdb->num_queries );
+	}
+
+	/**
+	 * @ticket 21760
+	 */
+	function test_invalidating_term_caches_should_fail_when_invalidation_is_suspended() {
```

Comment:
This seems to be more a test of `clean_term_cache()` than of `get_term_by()`. If the idea is that " `get_term_by()` should trust the cache no matter what", then it might be worth updating the term (name=Taco) while cache invalidation is suspended, and then showing that `get_term_by( ... )->name != 'Taco'`